src/freenet/clients/fcp/ClientPut.java

```diff
@@ -527,17 +530,19 @@ RequestStatus getStatus() {
 				total = (int) msg.getTotalBlocks();
 				min = (int) msg.getMinBlocks();
 				fetched = (int) msg.getFetchedBlocks();
+				latestSuccess = msg.getLatestSuccess();
 				fatal = (int) msg.getFatalyFailedBlocks();
 				failed = (int) msg.getFailedBlocks();
+				latestFailure = msg.getLatestFailure();
 				totalFinalized = msg.isTotalFinalized();
 			}
 		}
 		
-		return new UploadFileRequestStatus(identifier, persistence, started, finished, 
-				succeeded, total, min, fetched, fatal, failed, totalFinalized, 
-				lastActivity, priorityClass, finalURI, uri, failureCode,
-				failureReasonShort, failureReasonLong, getDataSize(), mimeType,
-				fnam, isCompressing());
```

Comment:
I assume you meant to comment this at the whole changed constructor function signature, not just at the single point where it is called here?
In case that is true, here's my reply:

The intentions of changing the signature like that were:
- Add the new parameter at a reasonable position in the parameter list, not just to the end. The "latestSuccess" is added after the "fetched" parameter, which is the number of successful blocks. The "latestFailure" is added after the "failed" parameter.
- To obey the line length limit, the parameters are shifted around to the next lines.

I assume what you complain about is the shifting of all parameters to obey the line length?
Do you see any other possibility of introducing new parameters in the middle without shifting while also obeying the line length limit?
I could of course insert a new line which contains nothing but the new parameter, but wouldn't that be ugly to the point where the current solution is preferable?